emote/typing.py

```diff
@@ -47,6 +52,20 @@ class EpisodeState(Enum):
     INITIAL = 3
 
 
+class BPStepScheduler:
+    def __init__(
+        self,
+        bp_step_begin: float,
+        bp_step_end: float,
+        value_min: float,
+        value_max: float,
+    ):
+        self.bp_step_begin = bp_step_begin
+        self.bp_step_end = bp_step_end
+        self.value_min = value_min
+        self.value_max = value_max
```

Comment:
Use a dataclass please. :-) Also, it *feels* like (a) this should be in something like `emote.extra.schedules`, and (b) it should have a evaluate_at or similar function? :-)

Reply:
Thanks for the great suggestion! Fixed it :)